cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -382,18 +380,18 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
                                                                         KafkaRebalanceAnnotation rebalanceAnnotation, RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
         switch (currentState) {
             case New:
-                return onNew(reconciliation, host, apiClient, rebalanceOptionsBuilder);
+                return onNew(reconciliation, host, apiClient, rebalanceOptionsBuilder, kafkaRebalance);
             case PendingProposal:
                 return onPendingProposal(reconciliation, host, apiClient, kafkaRebalance, rebalanceAnnotation, rebalanceOptionsBuilder);
             case ProposalReady:
                 return onProposalReady(reconciliation, host, apiClient, kafkaRebalance, rebalanceAnnotation, rebalanceOptionsBuilder);
             case Rebalancing:
                 return onRebalancing(reconciliation, host, apiClient, kafkaRebalance, rebalanceAnnotation);
             case Stopped:
-                return onStop(reconciliation, host, apiClient, rebalanceAnnotation, rebalanceOptionsBuilder);
+                return onStop(reconciliation, host, apiClient, rebalanceAnnotation, rebalanceOptionsBuilder, kafkaRebalance);
```

Comment:
Should we try to keep the same order of the parameters here? I see 4 methods hare which have already `reconciliation, host, apiClient, kafkaRebalance, ...`. So maybe you can insert it after the `apiClient` in the two new places as well?